### PR TITLE
Pocket: New base template (fixes #12302)

### LINKED ISSUE #12302: Pocket: two base templates
## Description

We put a quick fix in place to solve an issue in prod: https://github.com/mozilla/bedrock/pull/12301
But we'd like a more resilient solution.

Create two Pocket base templates: base and platform-base (or similar)
These templates should include header and footer (and should check if those elements are valid inside main tag, as in current implementation: https://github.com/mozilla/bedrock/blob/main/bedrock/pocket/templates/pocket/base.html#L44)

---

#### Success Criteria
- [ ] Two Pocket base templates
- [ ] Templates include header/footer
- [ ] Base template with mobile nav is tested for functionality

### PR DESCRIPTION
**Pull request by maureenlholland**
## One-line summary

This adds new base-platform template and moves header and footer out of content blocks so they can be adjusted on a page-by-page basis.

## Issue / Bugzilla link
https://github.com/mozilla/bedrock/issues/12302

## Testing

Mobile menu (Base template)

- [x] http://localhost:8000/en/about/
- [x] http://localhost:8000/en/contact-info/
- [x] http://localhost:8000/en/get-inspired/
- [x] http://localhost:8000/en/jobs/
- [x] http://localhost:8000/en/pocket-and-firefox/
- [x] http://localhost:8000/en/privacy/
- [x] http://localhost:8000/en/save-to-pocket/
- [x] http://localhost:8000/en/tos/

No mobile menu (Base template override)
- [x] http://localhost:8000/en
- [x] http://localhost:8000/en/add/

No mobile menu (Base platform template)
- [x] http://localhost:8000/en/chrome

note: **404, 500, home, and add** templates do not extend base-platform but **_do_** use the "platform" nav/footer
Any suggestions on a non-platform-specific name welcome

`BASE_POCKET_URL=http://localhost:8080 py.test -m pocket_mode --driver Firefox --html tests/functional/results.html tests/functional/`

### CODE REVIEW

File: tests/pages/pocket/base.py [hunk 1]
```diff
@@ -42,14 +44,16 @@ def URL_TEMPLATE(self):
 
     @property
     def navigation(self):
-        return self.Navigation(self)
+        if self.is_element_displayed(*self._platform_nav_locator):
```
Comment (maureenlholland) [inline tests/pages/pocket/base.py:47]: Not sure this is the right way to go about it, but some base templates override the provided base header/footer, so I wanted to allow the base test template to handle both.

File: bedrock/pocket/templates/pocket/base.html [hunk 2]
```diff
@@ -41,12 +41,22 @@
...
+    {% endblock %}
+
      {% block site_js %}
       {{ js_bundle('pocket-site') }}
      {% endblock %}
 
-     {% block js %}{% endblock %}
+     {% block js %}
+      {{ js_bundle('pocket-nav') }}
+     {% endblock %}
```
Comment (nathan-barrett) [inline bedrock/pocket/templates/pocket/base.html:60]: **Question**: If this .js file is only handling events in the nav, should this block be moved to the nav itself? Not sure what pattern is best for this situation

File: tests/pages/pocket/base.py (hunk 1 above)
Comment (nathan-barrett) [reply tests/pages/pocket/base.py:47]: seems pretty sensible to me!

**Review by nathan-barrett — APPROVED** [review]
r+ LGTM, great job on getting this work in 🎣

File: bedrock/pocket/templates/pocket/base.html (hunk 2 above)
Comment (alexgibson) [reply bedrock/pocket/templates/pocket/base.html:60]: Scripts should typically be inserted in the `js` block at the end of each page, else they can block rendering

**Pull request merged**